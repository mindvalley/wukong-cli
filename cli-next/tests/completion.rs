mod common;

#[test]
fn test_wukong_completion_help() {
    let cmd = common::wukong_raw_command()
        .arg("completion")
        .arg("--help")
        .assert()
        .success();

    let output = cmd.get_output();

    insta::assert_snapshot!(std::str::from_utf8(&output.stdout).unwrap());
}

#[test]
fn test_wukong_completion_bash() {
    let cmd = common::wukong_raw_command()
        .arg("completion")
        .arg("bash")
        .assert()
        .success();

    let output = cmd.get_output();

    insta::assert_snapshot!(std::str::from_utf8(&output.stdout).unwrap());
}

#[test]
fn test_wukong_completion_zsh() {
    let cmd = common::wukong_raw_command()
        .arg("completion")
        .arg("zsh")
        .assert()
        .success();

    let output = cmd.get_output();

    insta::assert_snapshot!(std::str::from_utf8(&output.stdout).unwrap());
}

#[test]
fn test_wukong_completion_fish() {
    let cmd = common::wukong_raw_command()
        .arg("completion")
        .arg("fish")
        .assert()
        .success();

    let output = cmd.get_output();

    insta::assert_snapshot!(std::str::from_utf8(&output.stdout).unwrap());
}
