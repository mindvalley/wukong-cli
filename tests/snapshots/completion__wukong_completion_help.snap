---
source: tests/completion.rs
expression: "std::str::from_utf8(&output.stdout).unwrap()"
---
Generate wukong cli completions for your shell to stdout

Usage: wukong completion [OPTIONS] <SHELL>

Arguments:
  <SHELL>  [possible values: bash, elvish, fish, powershell, zsh]

Options:
  -a, --application <APPLICATION>  Override the application name that the CLI will perform the
                                   command against. If the flag is not used, then the CLI will use
                                   the default application name from the config
  -v, --verbose...                 Use verbos output. More output per occurrence.
                                   
                                   By default, it'll only report errors.
                                   `-v` show warnings
                                   `-vv` show info
                                   `-vvv` show debug
                                   `-vvvv` show trace
  -q, --quiet...                   Do not print log message
  -h, --help                       Print help information

